dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/TemplatePage.java

```diff
@@ -129,7 +124,9 @@ public String init() {
             editMode = TemplatePage.EditMode.METADATA;
             template = templateService.find(templateId);
             template.setDataverse(dataverse);
-            template.setMetadataValueBlocks();
+
+            List<DatasetField> dsfForEdit = datasetFieldsInitializer.prepareDatasetFieldsForEdit(template.getDatasetFields(), dataverse);
```

Comment:
We must keep connection between template and datasetFields. Newly created datasetFields in this method will not have this connection. So in order for this to work, we must execute template.setDatasetFields()